lib/shelley/bench/restore-bench.hs

```diff
@@ -782,6 +853,16 @@ waitForWalletsSyncTo targetSync tr proxy walletLayer wids gp vData = do
     WalletLayer _ _ nl _ _ = walletLayer
     fst' (x,_,_) = x
 
+reportProgress :: NetworkLayer IO block -> Tracer IO (BenchmarkLog n) -> Percentage -> IO SlotNo -> IO ()
```

Comment:
this potentially can be factored out. I'm not sure how much spaghetti is best in case of benchmarks. I'd say less than tests, but cannot draw a line

Reply:
Less spaghetti than tests is a good rule of thumb.

My own policy is to touch existing code only lightly — if there's one or two obvious refactorings, I go for it, but if large parts are just suboptimal, I leave it as is. The reason is that code doesn't live very long, so the work I do on polishing existing code will be invalidated when the code kicks the bucket in 1-2 years.

Reply:
I think different parts of code can have different policies. But I agree on optimizing human cost in doubt.